packages/frontend/editor-ui/src/stores/workflows.store.ts

```diff
@@ -277,8 +277,8 @@ export const useWorkflowsStore = defineStore(STORES.WORKFLOWS, () => {
 
 	const getPastChatMessages = computed(() => Array.from(new Set(chatMessages.value)));
 
-	const connectionsByDestinationNode = computed(() =>
-		Workflow.getConnectionsByDestination(workflow.value.connections),
+	const connectionsByDestinationNode = computed(
+		() => cachedWorkflow?.connectionsByDestinationNode ?? {},
```

Comment:
Is this change supposed to prevent FE from generating ghost nodes? @netroy 

I wonder if it's safe to access cachedWorkflow without checking if it's up-to-date.

Reply:
no, this was meant to replace `Workflow.getConnectionsByDestination`, which doesn't exist, and probably should have never existed. We are leaking too much internal state here.

We need a separate PR to clean up connections properly on the frontend when nodes are deleted.